lib/common.js

```diff
@@ -19,5 +19,31 @@ exports.hexToBinary = function (str) {
   return Buffer.from(str, 'hex').toString('binary')
 }
 
+// HACK: Fix for WHATWG URL object not parsing non-standard URL schemes like
+// 'udp:'. Just replace it with 'http:' since we only need a few properties.
+//
+// Note: Only affects Chrome and Firefox. Works fine in Node.js, Safari, and
+// Edge.
+//
+// Note: UDP trackers aren't used in the normal browser build, but they are
+// used in a Chrome App build (i.e. by Brave Browser).
+//
+// Bug reports:
+// - Chrome: https://bugs.chromium.org/p/chromium/issues/detail?id=734880
+// - Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=1374505
+exports.parseUrl = function (str) {
+  const url = new URL(str.replace(/^udp:/, 'http:'))
+
+  if (str.match(/^udp:/)) {
```

Comment:
Nit: Since it's just being used to get a boolean, I prefer to use `/^udp:/.test(str)`.